### DISCUSSION TITLE
ZXing for SkiaSharp

### DISCUSSION THREAD

**Discussion by ndenkha** [micjahn/ZXing.Net#389] [Q&A]
Hi,

I found this package ZXing.Net.Bindings.SkiaSharp version 0.16.12, but it points to the same ZXing.Net which relies on GDI+.
They both seem to be the same ZXing.Net and support for SkiaSharp.

I've modified all my GDI+ to SkiaSharp, and I use ZXing.Net to read QR codes, and there is this call that relies on System.Drawing.Bitmap, which I can't find a corresponding SkiaSharp SKBitmap for it.
LuminanceSource lumSource = new BitmapLuminanceSource(gdiBitmap);

Any clarification of this other package and help how I can use ZXing.Net with SkiaSharp will be appreciated.

**Comment by micjahn** [comment]
It should be very easy.
Add the package ZXing.Net.Bindings.SkiaSharp to your project.
Create your reader instance, take your SKBitmap instance and call the method Decode:
```
var reader = new ZXing.SkiaSharp.BarcodeReader();
var skBitmap = new SKBitmap...
var result = reader.Decode(skBitmap);
```
The package contains the following additional classes of you need them:
ZXing.SkiaSharp.SKBitmapLuminanceSource
ZXing.SkiaSharp.BarcodeWriter 
ZXing.SkiaSharp.Rendering.SKBitmapRenderer

**Reply by ndenkha** [reply]
Excellent! so there is no more need for LuminanceSource.
Thank you!

**Reply by yassin-ajanif** [reply]
the same topic but with a different issue.
 i installed the package ZXing.Net.Bindings.SkiaSharp package but i couldn't  know how to save a generated barcode image using skiasharp the code you see is using system.drawing  which is not working for me with avalonia UI template.                    
                                                                                                                                                                                 
private static bool GenerateBarcode(double widthInCentimeters)
    {
        // Assuming 100 DPI (dots per inch) for simplicity
        int dpi = 100;

        // Calculate width and height in pixels
        int widthInPixels = (int)(widthInCentimeters * dpi / 2.54);
        int heightInPixels = widthInPixels / 2; // Assuming 1:2 aspect ratio

        string text = DateTime.Now.ToString("yyyyMMddHHmmssfff");

        BarcodeWriter writer = new BarcodeWriter
        {
            Format = BarcodeFormat.CODE_128,
            Options = new EncodingOptions
            {
                Height = heightInPixels,
                Width = widthInPixels
            }
        };
        Bitmap barcodeBitmap = writer.Write(text);

        // Get desktop path
        string desktopPath = Environment.GetFolderPath(Environment.SpecialFolder.Desktop);

        // Save the generated barcode image to the desktop path
        string filePath = Path.Combine(desktopPath, $"{text}.png");
        barcodeBitmap.Save(filePath, System.Drawing.Imaging.ImageFormat.Png);

        return true;
    }
